include/tscore/Scalar.h

```diff
@@ -893,6 +893,20 @@ Scalar<N, C, T>::minus(Counter n) const -> self
   return {_n - n};
 }
 
+template <intmax_t N, typename C>
```

Comment:
I think this logic deserves at least a comment.

Reply:
Based on our discussion, I went with a different implementation which hopefully is a bit clearer.